### PR TITLE
bump dfx version to 0.13.0-beta.3 and update ubuntu version in Dockerfile

### PR DESCRIPTION
**Pull request by mraszyk**
Bump dfx version to have the latest dfx features, e.g., dfx sns download and creating canisters via boundary nodes on testnets, available in SNS end-to-end tests.

### CODE REVIEW

**Review by lmuntaner — COMMENTED** [review]
Aren't there any breaking changes?

Did you try deploying to a testnet and still works?

**Comment by mraszyk** [comment]
> Aren't there any breaking changes?
> 
> Did you try deploying to a testnet and still works?

Looks good to me: https://qsgjb-riaaa-aaaaa-aaaga-cai.large05.testnet.dfinity.network/

File: dfx.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "dfx": "0.12.1",
```
Comment (peterpeterparker) [inline dfx.json:2]: What are the advantages or requirement to use dfx beta?

Comment (mraszyk) [reply dfx.json:2]: actually dfx is releasing soon, so I can wait a bit and update to 0.13.0 later today

Comment (mraszyk) [reply dfx.json:2]: the main motivation was to not switch between 0.12.1 and 0.13 in sns end-to-end tests (that need the latest features not available in 0.13)

Comment (peterpeterparker) [reply dfx.json:2]: That would be appreciated from my side if we can stick to official releases.

I work locally with NNS-dapp but also in other projects. 

dfx has currently no easy way to switch between versions - except reinstalling the all dfx - and I would be happy to not bump all my projects to beta.

i.e. by sticking to official releases my workflow is more smooth

That said, if there are advantages or features that are required in the beta for what you or we aim to develop, for sure this should not be a blocker.

Comment (mraszyk) [reply dfx.json:2]: I just got a confirmation that 0.13.0 releases today so I'll wait until then.

**Pull request merged**